Add installation instructions and citation to README (#143)
Happy-Algorithms-League/hal-cgp: README.md

The library provides a local search function that performs stochastic gradient descent on the values of `Parameter` nodes (cf. Topchy & Punch, 2001; Izzo et al., 2017).
See `examples/example_differential_evo_regression.py` for an example evolving an expression containing mathematical constants.

Installation
============

> To install the release version, simply execute

Reviewer: ```suggestion
You can install the latest relase via pip:
```

```shell script
pip install hal-cgp
```

> This library comes with some extra requirements, defined in `extra_requirements.txt`. For instance, if you want to parse an individual to a `sympy` expression or a `torch` object, you'll need to install the extra requirements by running

Reviewer: ```suggestion
This library depends on some optional packages defined in `extra_requirements.txt`. These are necessary, for example, to compile an individual to a SymPy expression or a PyTorch class. You can install all extra requirements via:
```

```shell script
pip install hal-cgp[all]
```

> To install the current master, run

Reviewer: ```suggestion
The adventurous can install the most recent development version directly from our master branch (don't use this in production unless there are good reasons!):
```
